docs/Inleiding.md

```diff
@@ -0,0 +1,43 @@
+# Inleiding
+
+## Historie
+
+VNGRealisatie is enkele jaren geleden begonnen met het inzetten van Imvertor als tool om de kwaliteit en de consistentie van de informatiemodellen te controleren. Tijdens dit traject rol ontstond het idee om op basis het informatiemodel en een omzettings-algoritme de basisentiteiten voor StUF te genereren. Toen dit realiseerbaar bleek te zijn bleek dat daar veel consistentie-voordelen aan zaten. Om die voordelen door te trekken naar de daadwerkelijke koppelvlakken is vervolgens ook een methodiek ontwikkeld om de koppelvlakken zelf te modelleren en genereren.
+Dit traject viel samen met de verschuiving van de focus binnen VNG Realisatie van StUF-XML naar Rest/Json-API's.
+Om deze aanpak toepasbaar te maken voor zowel Stuf-XML als Rest/Json API's is er een drielagen-model opgezet. In de afgelopen 2 jaar is de focus volledig verlegd naar het genereren van Rest/Json API-specificaties.
+
+Let wel, het eindproduct van dit traject is een API-specificatie en geen werkende software.  
+
+## Wat is Model Driven Design in de context van VNGR
+
+De Model Driven Design aanpak die door VNG Realisatie is ontwikkeld is gebaseerd op de visie dat het definiëren van uitwisselingsstandaarden begint bij het vastleggen van de semantiek en samenhang van de gegevens in een informatiemodel. Dit informatiemodel dien opgesteld te worden conform het [MetaModel Informatiemodellen](https://docs.geostandaarden.nl/mim/mim/).
```

Comment:
Leg je hierin de semantiek en samenhang van gegevens vast of leg je daarin juist de semantiek en samenhang tussen objecten in de realiteit vast? Het gaat volgens mij bij een SIM (of CIM) niet om de gegevens maar dat je een voor het op te lossen probleem zo compleet mogelijk beeld van de werkelijkheid modelleert. Dat doe je door die werkelijkheid vast te leggen in objecten en die objecten hebben toevallig kenmerken. Volgens mij moet je hier dus nog niet spreken van gegevens. Dat neigt nl. al naar een UGM.

Reply:
Ben ik het niet mee eens Robert. Een informatiemodel is geen modellering van de werkelijkheid. Het is het modelleren van de relevante gegevens van de eigenschappen van objecten in de werkelijkheid die je wilt vastleggen. 
Voorbeeld: Een informatieobject kan een brief zijn. Er is nergens vastgelegd op wat voor papier die brief is gedrukt en welke kleur dat papier heeft omdat dat voor ons niet relevant is. In de werkelijkheid bestaan die aspecten dus wel degelijk. 

Bij het maken van een informatiemodel kies je dus bij uitstek al voor het gegevens-perspectief. Een foto of schilderij is ook een model van de werkelijkheid (en misschien wel completer dan een informatie-model) maar daar kunnen we geen genormaliseerd gegevensmodel van maken om gestructureerde gegevens vast te leggen in een database.

Reply:
Klopt en ik denk dat we het ook wel met elkaar eens zijn alleen de wijze waarop we dat verwoorden verschilt. Als jij zegt

> Het is het modelleren van de relevante gegevens van de eigenschappen van objecten in de werkelijkheid die je wilt vastleggen.

dan komt dat n.m.m. overeen met mijn

> een voor het op te lossen probleem zo compleet mogelijk beeld van de werkelijkheid modelleert

ik zeg daar dus niet dat je de werkelijkheid helemaal modelleert. Alleen die delen die nodig zijn om het probleem op te lossen. Het gebruik van het woordje 'gegevens' neigt n.m.m. al te veel naar een UGM. In een SIM modelleer je de karakteristieken van een object nog niet de gegevens de je daarover gaat vastleggen. De uiteindelijke gegevens kunnen heel anders zijn dan de gemodelleerde karakteristieken.
Ik ben bang dat als je bij het maken van een SIM al te veel denkt in gegevens je belangrijke karakteristieken kunt missen.

Reply:
Ik wil er geen ellenlange discussie van maken, maar bij alle attributen in een informatiemodel leggen we al vast of het een string, een integer, een enumeratie of iets dergelijks is. Dat zijn echt gegevens-aspecten. 

Hoe denk jij dat ik dit dan zou moeten formuleren ?

Reply:
We moeten hier inderdaad niet te lang over discussiëren en ik wil er ook geen halszaak van maken maar zelf zou ik veel liever spreken over karakteristieken of properties i.p.v. over gegevens.